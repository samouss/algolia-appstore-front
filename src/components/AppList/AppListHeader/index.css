.AppListHeader {
  min-height: 11px;
  margin-bottom: 4px;

  @media (--tablet) {
    min-height: 15px;
  }
}

.AppListHeader__Stat {
  font-size: 0.75rem;
  text-align: right;
  font-style: italic;
  color: #b9b9b9;

  @media (--tablet) {
    font-size: 0.85rem;
  }
}

.AppListHeader__StatHits {
  font-weight: 500;
  font-size: 0.85rem;

  @media (--tablet) {
    font-size: 0.95rem;
  }
}
