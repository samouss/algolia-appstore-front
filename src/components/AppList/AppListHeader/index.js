import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const AppList = ({
  nbHits,
  processingTimeMS,
  isInitialLoad,
}) => (
  <div styleName="AppListHeader">
    {!isInitialLoad && (
      <p styleName="AppListHeader__Stat">
        <span styleName="AppListHeader__StatHits">{nbHits.toLocaleString()} results </span>
        <span>found in </span>
        <span>{processingTimeMS}ms</span>
      </p>
    )}
  </div>
);

AppList.propTypes = {
  nbHits: PropTypes.number.isRequired,
  processingTimeMS: PropTypes.number.isRequired,
  isInitialLoad: PropTypes.bool.isRequired,
};

export default AppList;
