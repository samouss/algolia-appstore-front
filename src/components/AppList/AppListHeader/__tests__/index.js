import React from 'react';
import { shallow } from 'enzyme';
import AppListHeader from '../index';

describe('<AppListHeader />', () => {
  const defaultProps = {
    nbHits: 500,
    processingTimeMS: 5,
    isInitialLoad: false,
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <AppListHeader
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render on initial load', () => {
    const props = {
      ...defaultProps,
      isInitialLoad: true,
    };

    const component = shallow(
      <AppListHeader
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
