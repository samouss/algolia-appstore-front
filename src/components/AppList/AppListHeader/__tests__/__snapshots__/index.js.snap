// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppListHeader /> expect to render 1`] = `
<div
  styleName="AppListHeader"
>
  <p
    styleName="AppListHeader__Stat"
  >
    <span
      styleName="AppListHeader__StatHits"
    >
      500
       results 
    </span>
    <span>
      found in 
    </span>
    <span>
      5
      ms
    </span>
  </p>
</div>
`;

exports[`<AppListHeader /> expect to render on initial load 1`] = `
<div
  styleName="AppListHeader"
/>
`;
